src/AutoRest.CSharp/Common/AutoRest/Plugins/CSharpGen.cs

```diff
@@ -144,8 +144,7 @@ await foreach (var file in project.GetGeneratedFilesAsync())
                 {
                     // Ignore any errors while trying to output crash information
                 }
-                await AutoRestLogger.Fatal($"Internal error in AutoRest.CSharp{ErrorHelpers.FileIssueText}\nException: {e.Message}\n{e.StackTrace}");
-                return false;
+                throw;
```

Comment:
it turns out this line is not working properly and the stack traces are swallowed somehow.
it is safe to throw the exception here because we have a try catch and print the stack traces clause in the place where calls this Execute method.